cmd/cri-containerd/options/options.go

```diff
@@ -18,38 +18,56 @@ package options
 
 import (
 	"flag"
+	"os"
 
+	"github.com/BurntSushi/toml"
 	"github.com/containerd/containerd"
 	"github.com/spf13/pflag"
 )
 
-// CRIContainerdOptions contains cri-containerd command line options.
-type CRIContainerdOptions struct {
+const configFilePathArgName = "config-file-path"
+
+// ContainerdConfig contains config related to containerd
+type ContainerdConfig struct {
+	// ContainerdSnapshotter is the snapshotter used by containerd.
+	ContainerdSnapshotter string `toml:"containerdsnapshotter"`
+	// ContainerdEndpoint is the containerd endpoint path.
+	ContainerdEndpoint string `toml:"containerdendpoint"`
+}
+
+// Config contains cri-containerd toml config
+type Config struct {
+	// contains config related to containerd
+	ContainerdConfig
 	// SocketPath is the path to the socket which cri-containerd serves on.
-	SocketPath string
+	SocketPath string `toml:"socketpath"`
 	// RootDir is the root directory path for managing cri-containerd files
 	// (metadata checkpoint etc.)
-	RootDir string
-	// PrintVersion indicates to print version information of cri-containerd.
-	PrintVersion bool
-	// ContainerdEndpoint is the containerd endpoint path.
-	ContainerdEndpoint string
-	// ContainerdSnapshotter is the snapshotter used by containerd.
-	ContainerdSnapshotter string
+	RootDir string `toml:"rootdir"`
 	// NetworkPluginBinDir is the directory in which the binaries for the plugin is kept.
-	NetworkPluginBinDir string
```

Comment:
Separate `CNIConfig`? So that we could config CNI separately:
```toml
[cni]
  plugin_bin_dir=""
  plugin_conf_dir=""
```

I think CNI and containerd are 2 things we want to have separate configurations for now.